core/src/main/java/org/apache/calcite/plan/Convention.java

```diff
@@ -66,6 +67,15 @@ default boolean useAbstractConvertersForConversion(RelTraitSet fromTraits,
     return true;
   }
 
+  /**
+   * Transform a RelBuilder to support creating physical RelNode with current convention.
+   * @param oldRelBuilder The original RelBuilder to be transformed.
+   * @return A new RelBuilder after transformation.
+   */
+  default RelBuilder transformRelBuilder(RelBuilder oldRelBuilder) {
```

Comment:
Why not simply having the following method:`default RelBuilderFactory getRelFactory()`. It is more intuitive and makes the existing callers simpler.  
Moreover, if we do this I think we can avoid adding the new getters/setters in `RelBuilder`.

Reply:
I also have this feeling, i can imagine how to use this method, initialize a logical builder first when invoke this method to switch to another builder, really weird ~ ! Returns a factory is more straight-forward.